phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterTableWithViewsIT.java

```diff
@@ -1174,21 +1175,16 @@ public void testDroppingIndexedColDropsViewIndex() throws Exception {
             }
             
             // verify index metadata was dropped
-            try {
-                viewConn.createStatement().execute("SELECT * FROM " + fullNameViewIndex1 );
-                fail("Index metadata should have been dropped");
-            } catch (TableNotFoundException e) {
-            }
-            
             pconn = viewConn.unwrap(PhoenixConnection.class);
-            view = pconn.getTable(new PTableKey(tenantId,  viewOfTable ));
+            view = pconn.getTableNoCache(viewOfTable);
+            assertEquals("Unexpected number of indexes ", 1, view.getIndexes().size());
+            assertEquals("Unexpected index ",  fullNameViewIndex2 , view.getIndexes().get(0).getName().getString());
+            assertNotEquals("Dropped index should not be in view metadata ",  fullNameViewIndex1 , view.getIndexes().get(0).getName().getString());
             try {
-                viewIndex = pconn.getTable(new PTableKey(tenantId,  fullNameViewIndex1 ));
+                viewIndex = pconn.getTableNoCache(fullNameViewIndex1);
```

Comment:
why is this change required?

Reply:
This is to make sure that the view index is fetched from the server and not from the client's cache. The test is verifying whether the view index was dropped on the server.